packages/framework/src/Component/Translation/JsFileExtractor.php

```diff
@@ -6,63 +6,48 @@
 use JMS\TranslationBundle\Model\Message;
 use JMS\TranslationBundle\Model\MessageCatalogue;
 use JMS\TranslationBundle\Translation\Extractor\FileVisitorInterface;
-use PLUG\JavaScript\JParser;
-use PLUG\JavaScript\JTokenizer;
-use Shopsys\FrameworkBundle\Component\Javascript\Parser\Translator\JsTranslatorCallParser;
 use SplFileInfo;
 use Twig_Node;
 
 class JsFileExtractor implements FileVisitorInterface
 {
-    /**
-     * @var \SplFileInfo
-     */
-    protected $file;
+    public const DUMP_FILE = 'translationsDump.json';
+
+    public const DEFAULT_MESSAGE_DOMAIN = 'messages';
 
     /**
      * @var \JMS\TranslationBundle\Model\MessageCatalogue
      */
     protected $catalogue;
 
-    /**
-     * @var \Shopsys\FrameworkBundle\Component\Javascript\Parser\Translator\JsTranslatorCallParser
-     */
-    protected $jsTranslatorCallParser;
-
-    /**
-     * @param \Shopsys\FrameworkBundle\Component\Javascript\Parser\Translator\JsTranslatorCallParser $jsTranslatorCallParser
-     */
-    public function __construct(JsTranslatorCallParser $jsTranslatorCallParser)
-    {
-        $this->jsTranslatorCallParser = $jsTranslatorCallParser;
-    }
-
     /**
      * @param \SplFileInfo $file
      * @param \JMS\TranslationBundle\Model\MessageCatalogue $catalogue
      */
     public function visitFile(SplFileInfo $file, MessageCatalogue $catalogue)
     {
-        if (substr($file, -3) !== '.js') {
+        if (substr($file, -strlen(static::DUMP_FILE)) !== static::DUMP_FILE) {
             return;
         }
 
-        $this->file = $file;
-        $this->catalogue = $catalogue;
+        $translationDumpContent = file_get_contents($file);
+        if ($translationDumpContent === false) {
+            return;
+        }
 
-        $filename = $file->getRealPath();
-        $contents = file_get_contents($filename);
+        $translationsDump = json_decode($translationDumpContent, true);
 
-        try {
-            $this->parseFile($contents);
-        } catch (\PLUG\parsing\ParseError $ex) {
-            throw new \Shopsys\FrameworkBundle\Component\Translation\Exception\ExtractionException(
-                $ex->getMessage() . "\n" . 'in file ' . $this->file->getRealPath()
-            );
-        } catch (\Shopsys\FrameworkBundle\Component\Javascript\Parser\Exception\JsParserException $ex) {
-            throw new \Shopsys\FrameworkBundle\Component\Translation\Exception\ExtractionException(
-                $ex->getMessage() . ' in file ' . $this->file->getRealPath()
+        foreach ($translationsDump as $translation) {
+            $message = new Message(
+                $translation['id'],
+                array_key_exists('domain', $translation) ? $translation['domain'] : static::DEFAULT_MESSAGE_DOMAIN
```

Comment:
_nitpick_ 🤓 

You can use `$translation['domain'] ?? static::DEFAULT_MESSAGE_DOMAIN`

Reply:
also, the domain key is the only one that is optional?

Reply:
In this case is domain only one optional key. Dumped object has one more optional key - locale, but it is not use in this class. Nitpick: 3a0f789139278d5e60e08320f03565a40150ddc7